mona_openai/endpoints/chat_completion.py

```diff
@@ -0,0 +1,222 @@
+"""
+The Mona wrapping code for OpenAI's Completion API.
+"""
+from copy import deepcopy
+from functools import wraps
+
+from ..util.oop_util import create_combined_object
+from ..analysis.privacy import get_privacy_analyzers
+from ..analysis.profanity import get_profanity_prob, get_has_profanity
+from ..analysis.textual import get_textual_analyzers
+from .endpoint_wrapping import OpenAIEndpointWrappingLogic
+
+CHAT_COMPLETION_CLASS_NAME = "ChatCompletion"
+
+
+def _get_choices_texts(response):
+    return tuple(
+        (choice["message"]["content"] for choice in response["choices"])
+    )
+
+
+def _get_prompt_texts(request):
+    return tuple(message["content"] for message in request["messages"])
+
+
+def _get_texts(func):
+    def wrapper(self, input, response):
+        return func(
+            self,
+            input["messages"][-1]["content"]
```

Comment:
can input["messages"] be an empty list?

Reply:
Nope. That's a requirement of the API.

Reply:
I mean - a user may do that, but then the OpenAI API will raise an exception, so this code will never be reached.